File: Problems/001. Two Sum/Solution.cpp
#include <unordered_map>
#include <vector>

// Runtime 2ms
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        vector<int> ans(2);

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            
            if (numMap.find(complement) != numMap.end()) {
                ans[0] = numMap[complement];
                ans[1] = i;
                return ans;
            }
            
            numMap[nums[i]] = i;
        }

        return ans;
    }
};

File: Problems/001. Two Sum/Solution.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_map: HashMap<i32, i32> = HashMap::new();
        
        for (i, num) in nums.iter().enumerate() {
            num_map.insert(*num, i as i32);
        }
        
        for (i, num) in nums.iter().enumerate() {
            let complement = target - num;

            if let Some(&index) = num_map.get(&complement) {
                if index != i as i32{
                    return vec![i as i32, index];
                }
            }
        }

        vec![]
    }
}